bson/src/main/org/bson/codecs/pojo/ClassModel.java

```diff
@@ -44,13 +46,13 @@
         this.name = clazz.getSimpleName();
         this.type = clazz;
         this.hasTypeParameters = clazz.getTypeParameters().length > 0;
-        this.propertyNameToTypeParameterMap = propertyNameToTypeParameterMap;
+        this.propertyNameToTypeParameterMap = new HashMap<String, TypeParameterMap>(propertyNameToTypeParameterMap);
```

Comment:
Can we just use `Collections.unmodifiableMap(propertyNameToTypeParameterMap)` that will enforce immutability

Reply:
Done